engine/sound/src/sound.cpp

```diff
@@ -320,16 +318,32 @@ namespace dmSound
             return r;
         }
 
+        float    master_gain = params->m_MasterGain;
+        uint32_t max_sound_data = params->m_MaxSoundData;
+        uint32_t max_sources = params->m_MaxSources;
+        uint32_t max_instances = params->m_MaxInstances;
+        uint32_t sample_frame_count = params->m_FrameCount; // 0 means, use the defaults
+
+        if (config)
+        {
+            master_gain = dmConfigFile::GetFloat(config, "sound.gain", 1.0f);
+            max_sound_data = (uint32_t) dmConfigFile::GetInt(config, "sound.max_sound_data", (int32_t) max_sound_data);
+            max_sources = (uint32_t) dmConfigFile::GetInt(config, "sound.max_sound_sources", (int32_t) max_sources);
+            max_instances = (uint32_t) dmConfigFile::GetInt(config, "sound.max_sound_instances", (int32_t) max_instances);
+            sample_frame_count = (uint32_t) dmConfigFile::GetInt(config, "sound.sample_frame_count", (int32_t) sample_frame_count);
+        }
+
         HDevice device = 0;
         OpenDeviceParams device_params;
+
         // TODO: m_BufferCount configurable?
         device_params.m_BufferCount = SOUND_OUTBUFFER_COUNT;
-        device_params.m_FrameCount = params->m_FrameCount;
+        device_params.m_FrameCount = sample_frame_count; // May be 0
```

Comment:
The OpenSL version uses the frame count already, when allocating internal memory.
We might separate it into two steps OpenDevice + InitializeDevice, as we could then get the sample rate from the device, in between those calls. For now, the OpenSL device does an extra step internally.